core/Concurrency/Lock.php

```diff
@@ -85,14 +99,14 @@ public function acquireLock($id, $ttlInSeconds = 60)
             // Lock key might be too long for DB column, so we hash it but leave the start of the original as well
             // to make it more readable
             $md5Len = 32;
-            $this->lockKey = mb_substr($id, 0, self::MAX_KEY_LEN - $md5Len - 1) . md5($id);
+            $this->lockKey = mb_substr($this->lockKey, 0, self::MAX_KEY_LEN - $md5Len - 1) . md5($id);
```

Comment:
Note: This actually fixes a bug that seems to have existed for quite some while. Before the code was actually throwing away the `lockKeyStart` part, which made it impossible for `getAllAcquiredLockKeys` to find the key anymore.